plugins/pulp_docker/plugins/distributors/publish_steps.py

```diff
@@ -250,28 +270,36 @@ class RedirectFileStep(publish_step.PublishStep):
     """
     This step creates the JSON file that describes the published repository for Crane to use.
     """
-    def __init__(self, app_publish_location):
+    def __init__(self, app_publish_location, redirect_data):
         """
         Initialize the step.
 
         :param app_publish_location: The full path to the location of the JSON file that this step
                                      will generate.
         :type  app_publish_location: basestring
+
+        :param redirect_data: Dictionary of tags and digests that manifests schema version 2
+                              reference
+        :type redirect_data:  dict
+
         """
         super(RedirectFileStep, self).__init__(step_type=constants.PUBLISH_STEP_REDIRECT_FILE)
         self.app_publish_location = app_publish_location
+        self.redirect_data = redirect_data
 
     def process_main(self):
         """
         Publish the JSON file for Crane.
         """
         registry = configuration.get_repo_registry_id(self.get_repo(), self.get_config())
         redirect_url = configuration.get_redirect_url(self.get_config(), self.get_repo(), 'v2')
+        schema2_data = self.redirect_data[2]
 
         redirect_data = {
-            'type': 'pulp-docker-redirect', 'version': 2, 'repository': self.get_repo().id,
+            'type': 'pulp-docker-redirect', 'version': 3, 'repository': self.get_repo().id,
             'repo-registry-id': registry, 'url': redirect_url,
-            'protected': self.get_config().get('protected', False)}
+            'protected': self.get_config().get('protected', False),
+            'schema2_data': list(schema2_data)}
```

Comment:
i know we already discussed that, i do not have any strong preference here, but i think, less data we need to work with and parse better is. I would like to hear @mhrivnak opinion as well on this - do we want to write also schema1_data and make it available to Crane?